kexec.c

```diff
@@ -693,6 +703,34 @@ static void exec_kexec(void)
 	exit(1);
 }
 
+static void do_file_load(char *kernel, char *initrd, char *cmdline)
+{
+	int kernel_fd = -1;
+	int initrd_fd = -1;
+	int ret = 0;
+	unsigned long cmdline_len;
+	unsigned long flags = 0;
+
+	/* cmdline_len includes the '\0'. */
+	cmdline_len = (cmdline == NULL)? 0 : strlen(cmdline) + 1;
+
+	if (kernel)
+		kernel_fd = open(kernel, O_RDONLY);
+
+	if (initrd)
+		initrd_fd = open(initrd, O_RDONLY);
+	else
+		flags |= KEXEC_FILE_NO_INITRAMFS;
+
+	debug_printf("kernel_fd=%d initrd_fd=%d cmdline_len=%lu flags=%lu\n",
+			kernel_fd, initrd_fd, cmdline_len, flags);
+	debug_printf("cmdline=\"%s\"\n", cmdline);
```

Comment:
Thanks for catching this! Apparently gcc catches null for `%s`, so I hadn't even seen this in testing.

The syscall will ignore the pointer if `cmdline_len` is zero. As long as that is picked up properly, then it doesn't particularly matter if `cmdline` is set to an empty string (sounds like a good idea anyway, however).